File: Paragraph.cs
namespace BookModel
{
	public class Paragraph : IBookElement
	{
		public string Text { get; set; }
		private IAlignStrategy _textAlignment;

		public Paragraph(string text)
		{
			Text = text;
		}

		public void SetAlignStrategy(IAlignStrategy alignStrategy)
		{
			_textAlignment = alignStrategy;
		}

		public void Print()
		{
			if (_textAlignment != null)
			{
				_textAlignment.Render(this);
			}
			else
			{
				Console.WriteLine($"Paragraph: {Text}");
			}
		}

		// Leaf: no-op composite methods
		public void Add(IBookElement element) { }
		public void Remove(IBookElement element) { }
		public IBookElement Get(int index) { return null; }

		public override string ToString()
		{
			return $"Paragraph: {Text}";
		}
	}
}

File: src/main/java/Paragraph.java
public class Paragraph implements Element {
    private String text;
    private AlignStrategy alignStrategy; 

    public Paragraph(String text) {
        this.text = text;
    }

    public String getText() {
        return text;
    }

    public void setAlignStrategy(AlignStrategy strategy) {
        this.alignStrategy = strategy;
    }

    @Override
    public void print() {
        if (alignStrategy != null) {
            alignStrategy.render(this, "Default Context"); 
        } else {
            System.out.println("Paragraph: " + text);
        }
    }

    @Override
    public void add(Element element) {
        throw new UnsupportedOperationException();
    }
    @Override
    public void remove(Element element) {
        throw new UnsupportedOperationException();
    }
    @Override
    public Element get(int index) {
        throw new UnsupportedOperationException();
    }
}